foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/copying/RecordCopyPolicy.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Oracle - initial API and implementation from Oracle TopLink
+package org.eclipse.persistence.descriptors.copying;
+
+import org.eclipse.persistence.exceptions.DescriptorException;
+import org.eclipse.persistence.sessions.Session;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.RecordComponent;
+import java.util.ArrayList;
+
+/**
+ * <p><b>Purpose</b>: This is the copy policy for {@code java.lang.Record}.
+ * <p>
+ * It creates a copy by creating a new instance. As {@code java.lang.Record} if immutable
+ * all cloned attributes must be collected and passed to the constructor.
+ */
+public class RecordCopyPolicy extends AbstractCopyPolicy {
+    public RecordCopyPolicy() {
+        super();
+    }
+
+    @Override
+    public Object buildClone(Object domainObject, Session session) throws DescriptorException {
+        return cloneRecord((Record) domainObject);
+    }
+
+    @Override
+    public boolean buildsNewInstance() {
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "()";
```

Comment:
would it make sense to `default` this in the `CopyPolicy` interface? I see this exact impl used in other `CopyPolicy` implementations..